spring-integration-file/src/main/java/org/springframework/integration/file/splitter/FileSplitter.java

```diff
@@ -263,7 +265,7 @@ public void close() throws IOException {
 			public boolean hasNext() {
 				this.hasNextCalled = true;
 				try {
-					if (this.line == null && !this.done) {
+					if (!this.done && this.line == null) {
```

Comment:
Why did you swap conditions?

Reply:
Because one line below is:

```
boolean ready = !this.done && this.line != null;
```

And the code reading is easier when both expressions have conditions in the same order.